### DISCUSSION TITLE
Ordering food from restaurants

### DISCUSSION THREAD

**Discussion by amuradyan** [pmuradian/Garcon#4] [Ideas]
> Garçon is a slack bot designed for ~ ordering lunches ~ from [buy.am](https://buy.am/hy/) delivery service.

@pmuradian, I'd like to hear your thought on this as well. 

I'm interested, if, for example, it should have access to a payment method or should it have an account, and, if so, how should it login? Also, how do you see the business process, e.g. should it ask for permission to pay each time, or should it be trusted to pay, if allowed once?

**Comment by pmuradian** [comment]
For the beginning it should not be able to make payments, it should only make the order. You will pay for the delivery with cash. A buy.am account can be created for convenience. If there is a public API from buy.am, ordering food should be fairly simple, just an API call, otherwise I am not sure what to do.

**Comment by amuradyan** [comment]
I see. 

It just came to my mind, that _Garcon_ should somehow track the order state. As we all know, sometimes there are no okroshkas, so you replace those with tans, or choose to remove them from the order altogether. Sometimes you get a call and are told the wait will be in an hour and a half, so you cancel the order.
